cmd/adapter/config.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/logrange/logrange/pkg/forwarder"
+	"github.com/logrange/logrange/pkg/utils"
+	"github.com/logrange/range/pkg/transport"
+	"io/ioutil"
+)
+
+type (
+	gravity struct {
+		ApiListenAddr string
+		Kubernetes    *kube
+	}
+
+	kube struct {
+		Namespace              string
+		ForwarderConfigMapName string
+	}
+
+	// cronQuery describe a query, which should be called periodically
+	cronQuery struct {
+		// Query contains LQL which should be called periodically
+		Query string
+		// PeriodSec defines an interval in seconds to execute the Query
+		PeriodSec int
+	}
+
+	logrange struct {
+		CronQueries       []cronQuery
+		Partition         string
+		ForwarderTmplFile string
+		Kubernetes        *kube
+		Transport         *transport.Config
+	}
+
+	Config struct {
+		Gravity         *gravity
+		Logrange        *logrange
+		SyncIntervalSec int
+	}
+)
+
+func LoadCfgFromFile(path string) (*Config, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	cfg := &Config{}
+	err = json.Unmarshal(data, cfg)
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+//===================== config =====================
+
+func NewDefaultConfig() *Config {
+	return &Config{
+		Gravity: &gravity{
```

Comment:
It is not too common to export pure data fields when their type isn't - the callers outside the package will not be able to access any attributes anyways.

Reply:
this is intentional for JSON (un-)marshaling, added comment to the struct emphasize this fact...